osvimdriver/openstack/neutron/driver.py

```diff
@@ -14,36 +18,69 @@ def __init__(self, session):
     def __get_neutron_client(self):
         return self.__neutron_client
 
-    def get_network_by_id(self, network_id):
+    def get_network_by_id(self, network_id,driver_request_id=None):
         if network_id is None:
             raise ValueError('network_id must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with id %s', network_id)
-        result = neutron_client.show_network(network_id)
-        return result['network']
+        try:
+            external_request_id = str(uuid.uuid4())
+            common._generate_additional_logs('', 'sent', external_request_id, '',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/networks/' + network_id }, driver_request_id)
+            result = neutron_client.show_network(network_id)
+            common._generate_additional_logs(result, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : 200, 'status_reason_phrase' : 'ok'}, driver_request_id)  
+            return result['network']
+        except Exception as e:
+            common._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.status_code,'status_reason_phrase' : e.message }, driver_request_id)
+            raise e
 
-    def get_network_by_name(self, network_name):
+
+    def get_network_by_name(self, network_name,driver_request_id=None):
         if network_name is None:
             raise ValueError('network_name must be provided')
         neutron_client = self.__get_neutron_client()
         logger.debug('Retrieving network with name %s', network_name)
-        result = neutron_client.list_networks()
+        external_request_id = str(uuid.uuid4())
+        common._generate_additional_logs('', 'sent', external_request_id, '',
+                                       'request', 'http', {'method' : 'get', 'uri' : LOG_URI_PREFIX +'/networks' }, driver_request_id)
+        try:
+            result = neutron_client.list_networks()
+            common._generate_additional_logs(str(result), 'received', external_request_id, 'plain/text',
```

Comment:
I am expecting the library would return the values in dict/json. May be you can update the content type to application/json and here you have already converted the result to str.

Reply:
I have made changes.